src/modules/nfts/components/wizard/RoyaltiesCreators.tsx

```diff
@@ -0,0 +1,706 @@
+    const total = creators.reduce((totalShares, creator) => {
+      return totalShares + creator.share;
+    }, 0);
+
+    setTotalRoyaltiesShare(total);
+  }, [creators]);
+
+  // TODO: DRY this up
+  const applyToAll = () => {
+    const zeroedRoyalties = creators.filter((creator) => creator.share === 0);
+
+    if (totalRoyaltyShares === 0 || totalRoyaltyShares > 100 || zeroedRoyalties.length > 0) {
+      setShowErrors(true);
+      return;
+    }
+
+    form
+      .validateFields(['royaltiesPercentage', 'numOfEditions'])
+      .then(() => {
+        if (setDoEachRoyaltyInd) setDoEachRoyaltyInd(false);
+
+        if (formValues) {
+          const newFormValues = formValues.map((formValue) => {
+            if (!creators.length || !maxSupply || !royaltiesInput) {
+              throw new Error('No creators or max supply or royalties input');
+            }
+            formValue.properties = { creators, maxSupply };
+            formValue.seller_fee_basis_points = royaltiesInput;
+            return formValue;
+          });
+          dispatch({ type: 'SET_FORM_VALUES', payload: [...newFormValues] });
+          nextStep!();
+        } else {
+          throw new Error('No form values found');
+        }
+      })
+      .catch((err) => {
+        console.log('err', err);
+      });
+  };
+
+  // TODO: DRY this up
+  const next = () => {
+    form.validateFields(['royaltiesPercentage']).then(() => {
+      if (formValues) {
+        const currentNFTFormValue = formValues[index];
+        if (!creators.length || !maxSupply || !royaltiesInput) {
+          throw new Error('No creators or max supply or royalties input');
+        }
+
+        currentNFTFormValue.properties = { creators, maxSupply };
+        currentNFTFormValue.seller_fee_basis_points = royaltiesInput;
+
+        const formValuesCopy = [...formValues];
+        formValuesCopy[index] = { ...currentNFTFormValue };
+
+        setRoyaltiesInput(ROYALTIES_INPUT_DEFAULT);
+        form.setFieldsValue({ royaltiesPercentage: ROYALTIES_INPUT_DEFAULT });
+      } else {
+        throw new Error('No form values found');
+      }
+
+      nextStep!();
+    });
+  };
+
+  const updateCreator = (address: string, share: number) => {
+    const creatorIndex = creators.findIndex((creator) => creator.address === address);
+    setCreators([
+      ...creators.slice(0, creatorIndex),
+      { address, share },
+      ...creators.slice(creatorIndex + 1),
+    ]);
+  };
+
+  const addCreator = () => {
+    form
+      .validateFields(['addCreator'])
+      .then((values) => {
+        if (creators.length >= MAX_CREATOR_LIMIT) {
+          throw new Error('Max level of creators reached');
+        }
+        const newShareSplit = 98 / creators.length;
+        setCreators([
+          creators[0],
+          ...creators.slice(1).map((c) => ({ ...c, share: newShareSplit })),
+          { address: values.addCreator, share: newShareSplit },
+        ]);
```

Comment:
I believe I posted about this on another PR. No need to change but would like to discuss creators not being stored in antd state.

Reply:
It is in the from state actually, so shouldn't be too hard to refactor later :) 
![image](https://user-images.githubusercontent.com/10433154/142626214-1b47ff32-7557-43e5-95ca-619198f2d827.png)